### PR TITLE
URL fix script

### PR DESCRIPTION
**Pull request by cookersjs**
Fix for the addlinks problems.

Two things:

1) It is in the orthoinference code since a 'datafix' folder already existed here. We should consider moving this folder to somewhere more appropriate
2) Keep in mind this is just a hotfix, so we just need to review the efficacy of the fix

I'll share the files I created for this in Slack

### CODE REVIEW

File: orthoinference/src/main/java/org/reactome/datafix/KEGGAddLinksMycobacteriumH37RvLinkFix.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                identifierToPrefixMap.put(identifier, prefixWithIdentifier);
+            } else {
+                System.out.println("LOOKOUT!");
+            }
+        }
+
+        //TODO: Should display name be updated? Technically the identifier includes the prefix, but from Reactome's perspective we've never included the prefix for any of the KEGG identifiers.
+
+        // Now with the map, I get the reference database of interest. Then I get all referrals of it, which corresponds to the 178 DatabaseIdentifiers.
+        GKInstance keggMycoDBInst = (GKInstance) dba.fetchInstanceByAttribute(ReactomeJavaConstants.ReferenceDatabase, ReactomeJavaConstants.name, "=", "KEGG Gene (Mycobacterium tuberculosis H37Rv)").iterator().next();
```
Comment (jweiser) [inline orthoinference/src/main/java/org/reactome/datafix/KEGGAddLinksMycobacteriumH37RvLinkFix.java]: There's only one KEGG Gene (Mycobacterium tuberculosis) reference database in release_current, right?

**Comment by SolomonShorser-OICR** [comment]
Originally, 17 links were tested. 12 failed, 5 were ok. Does this process fix ALL identifiers such that all links will work correctly?

File: orthoinference/src/main/java/org/reactome/datafix/KEGGAddLinksMycobacteriumH37RvLinkFix.java (hunk 1 above)
Comment (cookersjs) [reply orthoinference/src/main/java/org/reactome/datafix/KEGGAddLinksMycobacteriumH37RvLinkFix.java]: Yes there is only one for H37Rv, and another for another strain called CDC1551

Comment (jweiser) [reply orthoinference/src/main/java/org/reactome/datafix/KEGGAddLinksMycobacteriumH37RvLinkFix.java]: Ah okay and the affected instances are all for H37Rv?

**Comment by cookersjs** [comment]
> Looks good to me as long as the test on a copy of the database worked. Did any of the print statement alerting you to anything unexpected come up during a test run?

@jweiser Nothing came up during the test run after I switched to using just the uniprot file outputs. When I initially parsed the KEGG download file it only had 113/178 identifiers, but uniprot had all 178

**Comment by cookersjs** [comment]
> Originally, 17 links were tested. 12 failed, 5 were ok. Does this process fix ALL identifiers such that all links will work correctly?

@SolomonShorser-OICR Yes the number of identifiers found in the uniprot files is 178, which matches the number of databaseIdentifiers associated with the M. tuberculosis H37Rv reference database instance

File: orthoinference/src/main/java/org/reactome/datafix/KEGGAddLinksMycobacteriumH37RvLinkFix.java (hunk 1 above)
Comment (cookersjs) [reply orthoinference/src/main/java/org/reactome/datafix/KEGGAddLinksMycobacteriumH37RvLinkFix.java]: Yes they are, sorry for belated answer. There was 178 DatabaseIdentifiers that are for H37Rv ReferenceDatabase (determined by solomon) and that was the exact number of referrals to the ReferenceDatabase that were updated

**Pull request merged**